meson.build

```diff
@@ -3,13 +3,38 @@ project('abseil-cpp', 'cpp',
   license : 'Apache-2.0',
   default_options : [
     'cpp_std=c++11',
+    'warning_level=0'
   ])
 
 cpp = meson.get_compiler('cpp')
 
 cpp_flags = [
-  '-Wno-sign-compare'
+  '-Wcast-qual',
+  '-Wconversion-null',
+  '-Wformat',
+  '-Wformat-security',
+  '-Wmissing-declarations',
+  '-Woverlength-strings',
+  '-Wpointer-arith',
+  '-Wundef',
+  '-Wunused-local-typedefs',
+  '-Wunused-result',
+  '-Wvarargs',
+  '-Wvla',
+  '-Wwrite-strings',
+  '-DNOMINMAX',
 ]
+
+if build_machine.cpu_family() in ['x86_64', 'amd64', 'AMD64']
```

Comment:
I don't thin this is what @jpakkane was suggesting.

Reply:
Ah, I see what you mean - yeap could just be ` x86_64` :+1:

Reply:
Also, this should be `host_machine` not `build_machine` IMO https://mesonbuild.com/Reference-manual.html#build_machine-object

Reply:
I didn't think of cross-compilation on this (as I was building it on [Odroid C2]) but I think your suggestion makes sense.

[Odroid C2]: https://wiki.odroid.com/odroid-c2/odroid-c2